### PR TITLE
Add Shovel olive type to write to a data store

### PR DESCRIPTION
**Pull request by apmasell**

### CODE REVIEW

File: shesmu-server/src/main/java/ca/on/oicr/gsi/shesmu/server/plugins/InvokeDynamicPitParameterDescriptor.java [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+            pitType -> {
+              // Okay, this requires deep digging into Java's generic type system. We want to insist
+              // that the final object we get is parameterised over <T>, the output rows we are
+              // providing. So, we expect to be given a type, not a class, that's parameterised by
+              // exactly one type variable.
+
+              // Now, we need to travel through the type hierarchy to determine if the type
+              // parameter we saw is actually the one to Pit. So, we want a relationship like
+              // this: X<T> extends Y<T>; Y<K> extends Pit<K>. We don't care if the layers in
+              // the middle rename of shuffle values. So, X<T> extends Y<T, Integer>; Y<A, B>
```
Comment (callunity) [inline shesmu-server/src/main/java/ca/on/oicr/gsi/shesmu/server/plugins/InvokeDynamicPitParameterDescriptor.java]: `rename of shuffle` -> `rename or shuffle`

File: implementation.md [hunk 2]
```diff
@@ -423,6 +423,56 @@ write the parameter into the action. `JsonParameter` is an implementation that
...
+
+To create a pit:
+
+1. Create a class _P_`<T>` that extends `Pit<T>`. `T` is the input row
+   type and will be used throughout.
+1. Decide on the schema and set up parameters to collect the readers.
+1. Implement the `consume` method that uses the readers to consume the data in
+   the provided input stream.
+
+To deliver a database to lives using `PluginFileType` or `PluginFile`:
```
[the remark this answers is not in the document]
Comment (apmasell) [reply implementation.md]: olives! :laughing:

File: implementation.md [hunk 3]
```diff
@@ -95,7 +95,7 @@ interlocking type systems:
 Shesmu's type system describe the types of every type that can be used in an
 olive. Because `Type` is used by the ASM library, Shesmu's types are
 represented by `Imyhat` objects (this is an Ancient Egyptian word for _mould of
```
Comment (iainrb) [inline implementation.md]: Not part of the PR, but should this be "mode of conduct"?

Comment (apmasell) [reply implementation.md]: Dunno. It was what the dictionary had. :man_shrugging:
